OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -132,8 +135,24 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
 		}
 
+		void UpdateTarget()
+		{
+			if (args.GuidedTarget.IsValidFor(args.SourceActor))
+				target = args.GuidedTarget.CenterPosition;
```

Comment:
I think in the original it would stop shooting (not suddenly, fading out) and probably stop tracking and target the last actor position that was in range until the tail reached that position.

I'll check later.